### PR TITLE
basic REST API

### PR DESCRIPTION
**Pull request by MilBia**
That's only basic server with few endpoint's. Full able to changes. Data types of variables in classes can by changed in next release.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -47,3 +47,12 @@ com_crashlytics_export_strings.xml
 crashlytics.properties
 crashlytics-build.properties
 fabric.properties
+
+# Idea
+.idea
```
Comment (pwr218154) [inline .gitignore]: Are you sure that we don' want to add idea projects? i mean, at some point this should be generic solution but for now we all depend on project configurations. IMO this should be removed.

**Comment by pwr218154** [comment]
I would suggest to split each model element to smaller patch. It is hard to review 800+ lines of code. Please, if not for review it is easier to fix on next pull request.

**Comment by MilBia** [comment]
I remove .idea form .gitignore and split all code to init commit and others with basics of each module. In next tasks I will polish that endpoints and add changes if they needed that.

File: .gitignore [hunk 2]
```diff
@@ -47,3 +47,9 @@ com_crashlytics_export_strings.xml
 crashlytics.properties
 crashlytics-build.properties
 fabric.properties
+
+*.iml
+
+# building
+target
+vendor
```
Comment (pwr218154) [inline .gitignore]: No line ending/
is this custom ignore for your app? rather put this inside project folder (server)

File: server/.idea/workspace.xml [hunk 3]
```diff
@@ -0,0 +1,332 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (pwr218154) [inline server/.idea/workspace.xml]: This file is custom configuration (it is even added to .gitignore, you must have used add using wildcard or force option)
please do not commit this.

File: server/.idea/.name [hunk 4]
```diff
@@ -0,0 +1 @@
+mimosa-soft
```
Comment (pwr218154) [inline server/.idea/.name]: maybe `sushi-server`?

**Review by pwr218154 — CHANGES_REQUESTED** [review]
i will not approve for time being, let somebody also check it

**Review by pwr218154 — APPROVED** [review]

**Pull request merged**